commitlint.config.js

```diff
@@ -0,0 +1,56 @@
+// commitlint.config.js
+module.exports = {
+    rules: {
+        // rules are defined as [ severity (2=error), "always"|"never", argument? ]. See
+        // https://commitlint.js.org/#/reference-rules?id=rules
+
+        // log-commit-info is used for debugging. It outputs the dict passed to a plugin.
+        // 'log-commit-info': [2, 'always'],
+        'breaking-change-indicator': [2, 'always'],
+        'type-enum': [2, 'always',
+            [
+                'feat',
+                'fix',
+                'dep',
+                'ci',
+                'refactor',
+                'doc',
+                'test',
+            ]
+        ]
+    },
+    parserPreset: {
+        parserOpts: {
+            // add matching groups for `!` before and after scope. Before scope is invalid, after scope is correct
+            // and enforces the "breaking-change" header (see below).
+            headerPattern: /^(\w*)(!?)(?:\(([\w\$\.\-\* ]*)\))?(!?)\: (.*)$/,
+            headerCorrespondence: ["type", "wrong_breaking", "scope", "breaking", "subject"],
+        },
+    },
+    plugins: [
+        {
+            rules: {
+                // log-commit-info is used for debugging. It outputs the dict passed to a plugin.
+                // 'log-commit-info': (input) => {
+                //     console.log(input)
+                //     return [true, "see above"]
+                // },
+                'breaking-change-indicator': ({wrong_breaking, breaking, scope, body, footer}) => {
+                    if (scope && wrong_breaking) {
```

Comment:
Sounds to me like you could use `dep!!: this is still valid` because we're not checking the rule if there's no scope?

Reply:
even better: because the type was not enforced, `dep!!` was just not parsed as type and commit lint said "ok, come in."

Now I'm enforcing the type, and reworked the regex, so it will only match the indicator in the "wrong" part if there's a scope defined at all.